backend/src/halo2_impl.rs

```diff
@@ -11,6 +11,28 @@ impl<T: FieldElement> BackendImpl<T> for Halo2Prover {
         Halo2Prover::new(degree)
     }
 
+    fn add_verification_key(
+        &mut self,
+        pil: &Analyzed<T>,
+        fixed: &[(String, Vec<T>)],
+        vkey: Vec<u8>,
```

Comment:
This argument seems read from a file. Passing it as a `Vec<u8>` seems inconsistent with the setup argument, that receives a `&mut dyn io::Read`. Not sure if it is comparable or should be uniformized.

Reply:
I tried it before as `&mut dyn io::Read` but it wouldn't compile, because Halo2 only reads from `impl io::Read` not dyn